lib/chef/knife/ec2_server_create.rb

```diff
@@ -1108,15 +1108,21 @@ def wait_for_direct_sshd(hostname, ssh_port)
       end
 
       def ssh_connect_host
-        @ssh_connect_host ||= if config[:server_connect_attribute]
-                                server.send(config[:server_connect_attribute])
-                              else
-                                if vpc_mode? && !config[:associate_public_ip]
-                                  server.private_ip_address
-                                else
-                                  server.dns_name || server.public_ip_address
-                                end
-                              end
+        if config[:server_connect_attribute]
+          connect_attribute = config[:server_connect_attribute]
+        else
+          if vpc_mode? && !config[:associate_public_ip]
+            connect_attribute = "private_ip_address"
+          else
+            connect_attribute = server.dns_name ? "dns_name" : "public_ip_address"
+          end
+        end
+
+        @ssh_connect_host ||= server.send(connect_attribute)
```

Comment:
If you're not going to wrap the conditional logic above based on if this instance variable is set, it doesn't need to be an instance variable because the only place it is used is a few lines down. The purpose of wrapping the logic is so that logic is only parsed once per run.